src/app/plugins/animal.plugin.ts

```diff
@@ -0,0 +1,189 @@
+import { Plugin } from '../../common/plugin';
+import { ChannelType, IContainer, IHttpResponse, IMessage } from '../../common/types';
+import { MessageEmbed } from 'discord.js';
+import ms from 'ms';
+
+export default class AnimalPlugin extends Plugin {
+  public commandName: string = 'animal';
+  public name: string = 'Animal Plugin';
+  public description: string = 'Get wild with a fraction of gods creatures';
+  public usage: string =
+    'animal <Species (optional)> <subSpecies(Optional)> | animal list (species or <species> or subSpecies)';
+  public override pluginAlias = ['wild', 'species'];
+  public permission: ChannelType = ChannelType.Public;
+  public override pluginChannelName: string = Constants.Channels.Public.Pets;
+
+  private _API_URL: string = 'https://api.fetchit.dev/';
+
+  private _validAnimals: Set<string> = new Set([]); // All animals and species
+  private _animals: ISubSpecies[] = []; // All species.
+
+  private _ANIMAL_UPD_THRESH: number = ms('1d');
+  private _LAST_UPD_TIME: number = 0;
+
+  constructor(public container: IContainer) {
+    super();
+    this._updateAnimalAPIData();
+  }
+
+  public execute(message: IMessage, args?: string[]) {
+    this._updateAnimalAPIData();
+    const input = args?.length ? args : [this._getRandomAnimal()];
+
+    // Checks if they want to know what species/ subspecies are available.
+    if (input[0].startsWith('list')) {
+      this._pickListType(message, input[1]);
+      return;
+    }
+
+    // Preps the lookup data, and checks if it's valid.
+    const animalLookUp: string = input.join('_');
+    if (!this._validAnimals.has(animalLookUp)) {
+      message.reply(`\`${input}\`is a invalid species/ sub-species.`);
+      return;
+    }
+
+    const searchTerms: string = this._createSearchTerm(animalLookUp, !!this._findAnimalsFromSubspecies(animalLookUp)?.species);
+
+    this.container.httpService
+      .get(`${this._API_URL}species/?${searchTerms}`)
+      .then(async (response: IHttpResponse) => {
+        // Notifies the user if there was a problem contacting the server
+        if (Math.floor(response.status / 100) !== 2) {
+          message.reply(
+            `Something seems to have happened with the connection to ${this._API_URL}.`
+          );
+          return;
+        }
+
+        const [animalData] = response.data.animal;
+        await message.reply('', { files: [animalData.Image], name: 'image.jpg' });
+      })
+      .catch((err) => {
+        this.container.loggerService.warn(err);
+      });
+  }
+
+  private _createSearchTerm(data: string, sub: boolean) {
+
+    const species = this._findAnimalsFromSubspecies(data) ?? this._findAnimalsFromSpecies(data);
+    const subSpecies = (data: string) => {
+      if (sub) {
+        return data;
+      } 
+      if (species.subSpecies.length > 0) {
+        return this._getRandomSubSpecies(species);
+      } 
+    };
+
+    return `name=${species.species}&subspecies=${subSpecies}`;
+
+  }
```

Comment:
I dont understand this. This could use either comments or a good refactor
or both

Reply:
Had a go at commenting and changing the name of the function. You'll have to let me know if that is good enough, cause I'm too lazy to refactor the first time around.